cardano-node/src/Cardano/Node/Run.hs

```diff
@@ -439,12 +432,19 @@ createDiffusionArguments NodeConfiguration {
 
 producerAddresses
   :: NetworkTopology
-  -> [(RelayAddress,  PeerAdvertise)]
+  -> ([(Int, Map RelayAddress PeerAdvertise)], [RelayAddress])
 producerAddresses nt =
   case nt of
     RealNodeTopology lrpg prp _ ->
-      concatMap (rootAddressToRelayAddress . localRoots) (groups lrpg)
-      ++ concatMap rootAddressToRelayAddress (map publicRoots prp)
+       ( map (\lrp -> ( valency lrp
+                      , Map.fromList $ rootAddressToRelayAddress
+                                     $ localRoots lrp
+                      )
+             )
+             (groups lrpg)
+      , concatMap (map fst . rootAddressToRelayAddress)
+                  (map publicRoots prp)
+      )
```

Comment:
I think this is still misaligned by one space :).

Reply:
I'll do it by hand then 😛